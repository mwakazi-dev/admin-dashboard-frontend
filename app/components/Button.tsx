"use client";

import React, { FC } from "react";
import { ButtonType } from "antd/es/button";

import StyledButton from "@/styles/Button";

interface Props {
  children: React.ReactNode;
  variant?: ButtonType;
  loading?: boolean;
  disabled?: boolean;
  onClick?: () => void;
}
const Button: FC<Props> = ({
  children,
  variant = "primary",
  loading = false,
  disabled = false,
  onClick = () => {},
}) => {
  return (
    <StyledButton
      type={variant}
      loading={loading}
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </StyledButton>
  );
};

export default Button;
