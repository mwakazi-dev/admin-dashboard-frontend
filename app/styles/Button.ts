"use client";

import { Button } from "antd";
import styled from "styled-components";

const StyledButton = styled(Button)`
  color: black;
  margin-top: 40px;
  margin-left: 30px;
  font-family: var(--font-lato), sans-serif;
`;

export default StyledButton;
