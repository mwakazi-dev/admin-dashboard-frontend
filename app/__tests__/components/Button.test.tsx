import React from "react";
import { render } from "@testing-library/react";
import Button from "@/components/Button";

describe("Button Component", () => {
  it("renders children correctly", () => {
    const { getByText } = render(<Button>Click Me</Button>);
    expect(getByText("Click Me")).toBeInTheDocument();
  });

  it("handles click event", () => {
    const onClick = jest.fn();
    const { getByText } = render(<Button onClick={onClick}>Click Me</Button>);
    getByText("Click Me").click();
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Button>Click Me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
