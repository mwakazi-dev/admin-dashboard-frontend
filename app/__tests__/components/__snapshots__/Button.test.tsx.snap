// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-tjsggz ant-btn-primary ant-btn-color-primary ant-btn-variant-solid sc-VHjGu cswNpC"
    type="button"
  >
    <span>
      Click Me
    </span>
  </button>
</DocumentFragment>
`;
